### PR TITLE
data grid on telephony section (first)

### PR DESCRIPTION
**Pull request by virgotopia**

### CODE REVIEW

File: client/app/telecom/telephony/billingAccount/administration/linesGroup/telecom-telephony-billing-account-administration-lines-group.html [hunk 1]
```diff
@@ -40,63 +39,35 @@ <h1 data-translate="telephony_lines_group_title"></h1>
...
+                        }">
+
+                                <button class="btn btn-link"
+                                        data-ng-click="LinesGroupCtrl.selectBillingAccount($row)"
+                                        data-ng-bind=":: $row.description || $row.billingAccount">
+                                </button>
+                                <span class="text-nowrap ml-2"
+                                      data-ng-if="$row.status !== 'enabled'"
+                                      data-ng-bind=":: '(' + $row.status + ')'">
+                                </span>
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/administration/linesGroup/telecom-telephony-billing-account-administration-lines-group.html]: From line 49 to 56, only one tab is needed, not two. (Indentation)

File: client/app/telecom/telephony/billingAccount/billing/repaymentHistory/telecom-telephony-billing-account-billing-repayment-history.html [hunk 2]
```diff
@@ -35,97 +35,30 @@ <h1 data-translate="telephony_group_billing_repayment_history_title"></h1>
...
+                                    data-property="date" data-sortable>
+                            <span class="text-nowrap"
+                                  data-ng-bind="$row.price.text.replace('-', '')">
+                            </span>
+                        </oui-column>
+
+                        <oui-action-menu data-align="end" data-compact>
+                            <oui-action-menu-item
+                                    data-text="{{'telephony_group_billing_repayment_history_list_download_csv' | translate}}"
+                                    on-click="RepaymentHistoryCtrl.download($row)">
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/billing/repaymentHistory/telecom-telephony-billing-account-billing-repayment-history.html]: Indent (or realign prop on `oui-action-menu-item`)

File: client/app/telecom/telephony/billingAccount/billing/tollfreeHistory/telecom-telephony-billing-account-billing-tollfree-history.html [hunk 3]
```diff
@@ -22,99 +22,31 @@ <h1 data-translate="telephony_group_billing_tollfree_history_title"></h1>
...
+                            <span class="text-nowrap"
+                                  data-ng-bind="$row.price.text.replace('-', '')">
+                            </span>
+                        </oui-column>
+
+                        <oui-action-menu data-align="end" data-compact>
+                            <oui-action-menu-item
+                                    data-text="{{'telephony_group_billing_tollfree_history_list_download_csv' | translate}}"
+                                    on-click="TollfreeHistoryCtrl.download($row)">
+                            </oui-action-menu-item>
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/billing/tollfreeHistory/telecom-telephony-billing-account-billing-tollfree-history.html]: Indent (or realign prop on `oui-action-menu-item`)

File: client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html [hunk 4]
```diff
@@ -206,103 +206,38 @@
...
-                                    data-translate="telephony_group_billing_dashboard_bill_empty">
-                                </td>
-                            </tr>
-                        </tbody>
-                    </table>
-                </div>
+
+                <oui-datagrid data-rows="DashboardCtrl.bills" data-page-size="5">
+                    <oui-column data-title="'telephony_group_billing_dashboard_bill_date' | translate" data-property="date" data-sortable="desc">
+                            {{$row.date | date}}
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html]: indent

File: client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html [hunk 5]
```diff
@@ -206,103 +206,38 @@
...
+                    </oui-column>
+                    <oui-column data-title="'telephony_group_billing_dashboard_bill_amount' | translate" data-property="price.text" data-sortable>
+                            {{$row.price.text}}
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html]: indent

File: client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html [hunk 6]
```diff
@@ -206,103 +206,38 @@
...
+                        <span class="text-capitalize"
+                              data-ng-bind="($row.status === 'done' ? 'yes' : 'no') | translate">
+                        </span>
+                    </oui-column>
+
+                    <oui-action-menu data-align="end" data-compact>
+                        <oui-action-menu-item
+                                data-text="{{'telephony_group_billing_dashboard_bill_download_pdf' | translate}}"
+                                on-click="DashboardCtrl.download($row, 'pdf')">
+                        </oui-action-menu-item>
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html]: Indent (or realign prop on `oui-action-menu-item`)

File: client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html [hunk 7]
```diff
@@ -206,103 +206,38 @@
...
+        
+                        <oui-action-menu-item
+                                data-text="{{'telephony_group_billing_dashboard_bill_download_csv_emited' | translate}}"
+                                on-click="DashboardCtrl.download($row, 'csv')">
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html]: Indent (or realign prop on `oui-action-menu-item`)

File: client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html [hunk 8]
```diff
@@ -206,103 +206,38 @@
...
+                        </oui-action-menu-item>
+
+                        <oui-action-menu-item
+                                data-text="{{'telephony_group_billing_dashboard_bill_download_csv_received' | translate}}"
+                                on-click="DashboardCtrl.download($row, 'received.csv')">
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html]: Indent (or realign prop on `oui-action-menu-item`)

File: client/components/telecom/telephony/abbreviatedNumbers/telecom-telephony-abbreviated-numbers.html [hunk 9]
```diff
@@ -77,97 +77,34 @@
...
+                  data-ng-bind="$row.surname"></span>
+        </oui-column>
+        <oui-column data-title="'telephony_abbreviated_numbers_name' | translate" data-property="name" data-sortable>
+            <span class="word-break"
+                  data-ng-bind="$row.name"></span>
+        </oui-column>
+        <oui-action-menu data-align="end" data-compact>
+            <oui-action-menu-item
+                    data-text="{{'edit' | translate}}"
+                    on-click="$ctrl.update($row)">
```
Comment (JeremyDec) [inline client/components/telecom/telephony/abbreviatedNumbers/telecom-telephony-abbreviated-numbers.html]: Indent (or realign prop on `oui-action-menu-item`)

File: client/components/telecom/telephony/abbreviatedNumbers/telecom-telephony-abbreviated-numbers.html [hunk 10]
```diff
@@ -77,97 +77,34 @@
...
+            </oui-action-menu-item>
+            <oui-action-menu-item
+                    data-text="{{'delete' | translate}}"
+                    on-click="$ctrl.remove($row)">
```
Comment (JeremyDec) [inline client/components/telecom/telephony/abbreviatedNumbers/telecom-telephony-abbreviated-numbers.html]: Indent (or realign prop on `oui-action-menu-item`)

File: client/app/telecom/telephony/billingAccount/administration/addGroup/telecom-telephony-billing-account-administration-add-group.controller.js [hunk 11]
```diff
@@ -18,10 +18,10 @@ angular.module("managerApp").controller("TelecomTelephonyBillingAccountAdministr
     }
 
     self.getDisplayedPrice = function () {
-        if (self.prices.withTax.value === 0) {
+        if (_.has(self, "prices.withTax.value") && self.prices.withTax.value === 0) {
             return $translate.instant("telephony_add_group_free");
         }
-        return self.prices.withTax.text;
+        return _.get(self, "a.b.c", "prices.withTax.text", "-");
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/administration/addGroup/telecom-telephony-billing-account-administration-add-group.controller.js]: -`"a.b.c"`?

File: client/app/telecom/telephony/billingAccount/administration/linesGroup/telecom-telephony-billing-account-administration-lines-group.html [hunk 12]
```diff
@@ -40,63 +39,34 @@ <h1 data-translate="telephony_lines_group_title"></h1>
...
-                    </div>
-                </div><!-- billing account -->
+
+                    <oui-datagrid data-rows="LinesGroupCtrl.billingAccounts.ids" data-row-loader="LinesGroupCtrl.fetchBillingAccountDetails($row)" page-size="5" >
+                        <oui-column data-title="'telephony_lines_group_telephony_group' | translate" data-property="status" data-sortable="asc"
+                        data-ng-class="{
+                            'text-muted': $row.status !== 'enabled' || $row.billingAccount === LinesGroupCtrl.billingAccounts.current
+                        }">
+
+                            <button class="btn btn-link"
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/administration/linesGroup/telecom-telephony-billing-account-administration-lines-group.html]: +`type="button"`?

File: client/app/telecom/telephony/billingAccount/billing/bill/telecom-telephony-billing-account-billing-bill.html [hunk 13]
```diff
@@ -15,154 +15,36 @@ <h1 data-translate="telephony_group_billing_bill_title"></h1>
...
-                 data-items-per-page="itemsPerPage"
-                 data-page-placeholder="{{'common_pagination_page' | translate: {currentPage:currentPage, nbPages:nbPages} }}"
-                 data-item-per-page-placeholder="{{'common_pagination_display' | translate}}">
-            </div>
-        </div>
-        <!-- /.widget-presentation -->
+
+            <oui-datagrid data-rows="BillingAccountBillCtrl.consumption.raw">
+                <oui-column data-title="'telephony_group_billing_bill_date' | translate" data-property="date" data-sortable="desc">
+                    {{$row.date | date}}
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/billing/bill/telecom-telephony-billing-account-billing-bill.html]: Prefer `data-ng-bind`?

File: client/app/telecom/telephony/billingAccount/billing/bill/telecom-telephony-billing-account-billing-bill.html [hunk 14]
```diff
@@ -15,154 +15,36 @@ <h1 data-translate="telephony_group_billing_bill_title"></h1>
...
+                </oui-column>
+                <oui-column data-title="'telephony_group_billing_bill_amount' | translate" data-property="price.value" data-sortable>
+                    {{$row.price.text}}
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/billing/bill/telecom-telephony-billing-account-billing-bill.html]: Prefer `data-ng-bind`?

File: client/app/telecom/telephony/billingAccount/billing/summary/telecom-telephony-billing-account-billing-summary.controller.js [hunk 15]
```diff
@@ -40,37 +31,13 @@ angular.module("managerApp").controller("TelecomTelephonyBillingAccountBillingSu
...
-    /* -----  End of HELPERS  ------*/
-
-
     /*= =====================================
     =            INITIALIZATION            =
     ======================================*/
-
-    function init () {
-        self.loading.init = true;
+    this.$onInit = function () {
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/billing/summary/telecom-telephony-billing-account-billing-summary.controller.js]: Missing one empty line separator.

File: client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html [hunk 16]
```diff
@@ -206,103 +206,35 @@
...
-                                    data-translate="telephony_group_billing_dashboard_bill_empty">
-                                </td>
-                            </tr>
-                        </tbody>
-                    </table>
-                </div>
+
+                <oui-datagrid data-rows="DashboardCtrl.bills" data-page-size="5">
+                    <oui-column data-title="'telephony_group_billing_dashboard_bill_date' | translate" data-property="date" data-sortable="desc">
+                        {{$row.date | date}}
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html]: Prefer `data-ng-bind`?

File: client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html [hunk 17]
```diff
@@ -206,103 +206,35 @@
...
+                    </oui-column>
+                    <oui-column data-title="'telephony_group_billing_dashboard_bill_amount' | translate" data-property="price.text" data-sortable>
+                        {{$row.price.text}}
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html]: Prefer `data-ng-bind`?

File: client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html [hunk 18]
```diff
@@ -206,103 +206,35 @@
...
-                        </tbody>
-                    </table>
-                </div>
+
+                <oui-datagrid data-rows="DashboardCtrl.bills" data-page-size="5">
+                    <oui-column data-title="'telephony_group_billing_dashboard_bill_date' | translate" data-property="date" data-sortable="desc">
+                        <span data-ng-bind="$row.date | date"></span>
+                    </oui-column>
+                    <oui-column data-title="'telephony_group_billing_dashboard_bill_amount' | translate" data-property="price.text" data-sortable>
+                        <span data-ng-bind="$row.price.text"></span>
```
Comment (whispyy) [inline client/app/telecom/telephony/billingAccount/dashboard/telecom-telephony-billing-account-dashboard.html]: (Very Minor): this line can be removed too.

File: client/app/telecom/telephony/service/consumption/outgoingFax/telecom-telephony-service-consumption-outgoingFax.html [hunk 19]
```diff
@@ -54,151 +54,35 @@
...
+        <oui-datagrid data-rows="$ctrl.consumption.sorted" data-page-size="10">
+                <oui-column data-title="'telephony_service_consumption_date' | translate"
+                            data-property="creationDatetime" data-sortable="desc">
+                    <span data-ng-bind="::$row.creationDatetime | date: 'dd/MM/yyyy'"></span>
+                    <span data-ng-bind="::$row.creationDatetime | date: 'HH:mm:ss'"></span>
+                </oui-column>
+
+                <oui-column data-title="'telephony_service_consumption_outgoing_fax_pages' | translate"
+                            data-property="pages">
+                    <span data-ng-bind="::$row.pages"></span>
```
Comment (whispyy) [inline client/app/telecom/telephony/service/consumption/outgoingFax/telecom-telephony-service-consumption-outgoingFax.html]: This line can be removed

File: client/app/telecom/telephony/service/consumption/incomingFax/telecom-telephony-service-consumption-incomingFax.html [hunk 20]
```diff
@@ -51,141 +51,33 @@
...
-             data-items="$ctrl.consumption.sorted"
-             data-paginated-items="$ctrl.consumption.paginated"
-             data-current-page="currentPage"
-             data-nb-pages="nbPages"
-             data-items-per-page="itemsPerPage"
-             data-page-placeholder="{{'common_pagination_page' | translate: {currentPage:currentPage, nbPages:nbPages} }}"
-             data-item-per-page-placeholder="{{'common_pagination_display' | translate}}">
-        </div>
+
+        <oui-datagrid data-rows="$ctrl.consumption.sorted" data-page-size="10">
```
Comment (neolitec) [inline client/app/telecom/telephony/service/consumption/incomingFax/telecom-telephony-service-consumption-incomingFax.html]: Why page-size is forced to 10? Is it normal?

File: client/app/telecom/telephony/billingAccount/billing/repaymentHistory/telecom-telephony-billing-account-billing-repayment-history.controller.js [hunk 21]
```diff
@@ -90,11 +66,8 @@ angular.module("managerApp").controller("TelecomTelephonyBillingAccountBillingRe
 
         return self.fetchFile(consumption).then(function (info) {
             $window.location.href = info.url;
-        }).catch(function (error) {
-            Toast.error([$translate.instant("telephony_group_billing_repayment_history_download_error"), (error.data && error.data.message) || ""].join(" "));
-            return $q.reject(error);
-        }).finally(function () {
-            consumption.downloading = false;
+        }, function (error) {
+            return new ToastError([$translate.instant("telephony_group_billing_repayment_history_download_error"), (error.data && error.data.message) || ""].join(" "));
```
Comment (Jisay) [inline client/app/telecom/telephony/billingAccount/billing/repaymentHistory/telecom-telephony-billing-account-billing-repayment-history.controller.js]: This is not how `ToastError` works. It takes as arguments an error object AND a translation key, not a string

File: client/app/telecom/telephony/billingAccount/billing/repaymentHistory/telecom-telephony-billing-account-billing-repayment-history.controller.js [hunk 22]
```diff
@@ -104,26 +77,17 @@ angular.module("managerApp").controller("TelecomTelephonyBillingAccountBillingRe
...
-                self.sortConsumption();
+                self.consumptionData = consumptions;
             });
-        }).catch(function (error) {
-            Toast.error([$translate.instant("telephony_group_billing_repayment_history_init_error"), (error.data && error.data.message) || ""].join(" "));
-            return $q.reject(error);
-        }).finally(function () {
-            self.loading.init = false;
+        }, function (error) {
+            return new ToastError([$translate.instant("telephony_group_billing_repayment_history_download_error"), (error.data && error.data.message) || ""].join(" "));
```
Comment (Jisay) [inline client/app/telecom/telephony/billingAccount/billing/repaymentHistory/telecom-telephony-billing-account-billing-repayment-history.controller.js]: This is not how `ToastError` works. It takes as arguments an error object AND a translation key, not a string

File: client/app/telecom/telephony/billingAccount/billing/repaymentHistory/telecom-telephony-billing-account-billing-repayment-history.controller.js [hunk 23]
```diff
@@ -38,25 +27,12 @@ angular.module("managerApp").controller("TelecomTelephonyBillingAccountBillingRe
                     consumption.priceValue = consumption.price ? consumption.price.value : null;
                 });
             });
+        }, function (err) {
+            self.consumptionData = [];
+            return new ToastError([$translate.instant("telephony_group_billing_repayment_history_download_error"), (err.data && err.data.message) || ""].join(" "));
```
Comment (Jisay) [inline client/app/telecom/telephony/billingAccount/billing/repaymentHistory/telecom-telephony-billing-account-billing-repayment-history.controller.js]: Be careful !!! This is not how `ToastError` works !!!
Here is the declaration : 
https://github.com/ovh-ux/ovh-manager-telecom/blob/3d1d22ba1564d43fa263f9758584513cd7dd4142/client/components/toast-error/toast-error.service.js#L1-L24
It takes as arguments an error object AND a translation key, not a string !!!
If you want a beautiful error message let like it was done before and simply use `Toast.error()` method and reject the error.

File: client/app/telecom/telephony/billingAccount/billing/tollfreeHistory/telecom-telephony-billing-account-billing-tollfree-history.controller.js [hunk 24]
```diff
@@ -36,26 +25,12 @@ angular.module("managerApp").controller("TelecomTelephonyBillingAccountBillingTo
                     consumption.priceValue = consumption.price ? consumption.price.value : null;
                 });
             });
+        }, function (err) {
+            self.consumptionData = [];
+            return new ToastError([$translate.instant("telephony_group_billing_tollfree_history_download_error"), (err.data && err.data.message) || ""].join(" "));
```
Comment (Jisay) [inline client/app/telecom/telephony/billingAccount/billing/tollfreeHistory/telecom-telephony-billing-account-billing-tollfree-history.controller.js]: This is not how `ToastError` works. It takes as arguments an error object AND a translation key, not a string

File: client/app/telecom/telephony/billingAccount/billing/tollfreeHistory/telecom-telephony-billing-account-billing-tollfree-history.controller.js [hunk 25]
```diff
@@ -84,15 +59,10 @@ angular.module("managerApp").controller("TelecomTelephonyBillingAccountBillingTo
...
-
         return self.fetchFile(consumption).then(function (info) {
             $window.location.href = info.url;
-        }).catch(function (error) {
-            Toast.error([$translate.instant("telephony_group_billing_tollfree_history_download_error"), (error.data && error.data.message) || ""].join(" "));
-            return $q.reject(error);
-        }).finally(function () {
-            consumption.downloading = false;
+        }, function (err) {
+            return new ToastError([$translate.instant("telephony_group_billing_tollfree_history_download_error"), (err.data && err.data.message) || ""].join(" "));
```
Comment (Jisay) [inline client/app/telecom/telephony/billingAccount/billing/tollfreeHistory/telecom-telephony-billing-account-billing-tollfree-history.controller.js]: This is not how `ToastError` works. It takes as arguments an error object AND a translation key, not a string

File: client/app/telecom/telephony/billingAccount/billing/tollfreeHistory/telecom-telephony-billing-account-billing-tollfree-history.controller.js [hunk 26]
```diff
@@ -102,26 +72,15 @@ angular.module("managerApp").controller("TelecomTelephonyBillingAccountBillingTo
...
-                self.sortConsumption();
+                self.consumptionData = consumptions;
             });
-        }).catch(function (error) {
-            Toast.error([$translate.instant("telephony_group_billing_tollfree_history_init_error"), (error.data && error.data.message) || ""].join(" "));
-            return $q.reject(error);
-        }).finally(function () {
-            self.loading.init = false;
+        }, function (err) {
+            return new ToastError([$translate.instant("telephony_group_billing_tollfree_history_init_error"), (err.data && err.data.message) || ""].join(" "));
```
Comment (Jisay) [inline client/app/telecom/telephony/billingAccount/billing/tollfreeHistory/telecom-telephony-billing-account-billing-tollfree-history.controller.js]: This is not how `ToastError` works. It takes as arguments an error object AND a translation key, not a string

**Pull request merged**